### ISSUE TITLE
Regex validation extension on string types

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#266]
Consider if some string schema types could have an extensible regex validator for site specific validations.

**Comment by yaleman** [comment]
Can probably be done as part of the struct rebuilds in #1987

**Comment by Firstyear** [comment]
Not really, it's about adding a whole new syntax type with custom regex rules.
